synapse/handlers/appservice.py

```diff
@@ -86,7 +87,7 @@ async def notify_interested_services(self, max_token: RoomStreamToken):
                     if not events:
                         break
 
-                    events_by_room = {}
+                    events_by_room: Dict[str, List[EventBase]] = {}
```

Comment:
I don't think this will work on Python 3.5:

```suggestion
                    events_by_room = {}. # type: Dict[str, List[EventBase]]
```

Reply:
/me sets fire to python 3.5